tests/sqlite/sqlite.rs

```diff
@@ -725,3 +725,19 @@ async fn concurrent_read_and_write() {
     read.await;
     write.await;
 }
+
+#[sqlx_macros::test]
+async fn test_query_with_progress_handler() -> anyhow::Result<()> {
+    let mut conn = new::<Sqlite>().await?;
+    conn.set_progress_handler(1, || false).await;
```

Comment:
As alluded to above, this doesn't test that the code behaves correctly when the callback is not zero-sized.